addons/hr_holidays/views/hr_leave_allocation_views.xml

```diff
@@ -146,6 +146,13 @@
                                 <div id="no_limit_label" class="oe_read_only"
                                     invisible="not id or date_to or state != 'confirm'">No limit</div>
                             </div>
+                            <div colspan="2" class="oe_row alert alert-danger my-2" invisible=" not date_from or not date_to or not (date_to &lt; date_from)">
+                                <span>The allocation start date is set after the finish date.</span>
```

Comment:
```py
not date_from or not date_to or not (date_to &lt; date_from)
```

your last not can be just removed if you change the order of check to strictly greater than